### PR TITLE
Add support to Trip parser

### LINKED ISSUE #40: Add support for Trip Parser API

### PR DESCRIPTION
**Pull request by tsolakoua**
#### Support added as: 
- Trip Parser Jobs: amadeus.travel.trip_parser_jobs.post(file)
- Trip Parser Status: amadeus.travel.trip_parser_jobs.status(response.data['id']).get()
- Trip Parser Results: amadeus.travel.trip_parser_jobs.status(response.data['id']).get()

####  In the class `TripParser` we have the methods:
- post(body)
- def status(self, job_id):
- def result(self, job_id):

#### The `travel/_encoder.py` contains the following methods: 
- `def from_file(file)`: encodes a file to base64 and then calls the `construct_post_body(base64)` to create the JSON dict input to the POST. It's added to the namespaces of `travel.py` and can be called as `amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_file(file))`
- `def from_base64(base64)`: A base64 format booking is given as input and returns a JSON dict body. It's added to the namespaces of `travel.py` and can be called as `amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_base64(base64))`
- `def construct_post_body(base64)`: called by the methods above to construct a JSON dict body

#### The `mixins/encoder.py` contains the following methods: 
- `def encode_file_to_base64(file)`: encodes a file to base64

#### Others:

- Updates docs at `README.str` and tests at `namespaces_spec.py`.

### CODE REVIEW

**Comment by tsolakoua** [comment]
@alnacle , @anthonyroux changes made. Thanks for your feedback!

File: README.rst [hunk 1]
```diff
@@ -279,6 +279,16 @@ List of supported endpoints
     # AI Generated Photos
     amadeus.media.files.generated_photos.get(category='MOUNTAIN')
 
+    # Trip Parser
+    # Encode to Base64 your booking confirmation file
+    response = amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_file(file))
```
Comment (anthonyroux) [inline README.rst]: file is actually the path to the file no? Should we specify which formats we support?

File: README.rst [hunk 2]
```diff
@@ -279,6 +279,16 @@ List of supported endpoints
...
+    # Alternatively you can use a Base64 format directly
+    response = amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_base64(base64))
+    # Get the parsing status of the process by jobId
```
Comment (anthonyroux) [inline README.rst]: Do we want to add the 3rd case in the examples (where we pass a JSON directly)?

File: amadeus/travel/_trip_parser_jobs.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            '{ "data": {
+             "type": "trip-parser-job",
+             "content": : "base64string" }}'
+
+        You can call our helper functions with your booking details
+        in a file or a Base64 format
+
+        .. code-block:: python
+            amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_file(file))
+            amadeus.travel.trip_parser_jobs.post(amadeus.travel.from_base64(file))
```
Comment (anthonyroux) [inline amadeus/travel/_trip_parser_jobs.py]: the param is not a file for base64

File: amadeus/travel/_trip_parser_jobs.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        .. code-block:: python
+
+            amadeus.travel.trip_parser_jobs.post(
+            '{ "data": {
+             "type": "trip-parser-job",
+             "content": : "base64string" }}'
+
+        You can call our helper functions with your booking details
+        in a file or a Base64 format
```
Comment (anthonyroux) [inline amadeus/travel/_trip_parser_jobs.py]: or the content encoded in base64*

File: specs/namespaces/namespaces_spec.py [hunk 5]
```diff
@@ -244,3 +254,15 @@
...
+        with it('.travel.trip_parser_jobs.status().get'):
+            self.client.travel.trip_parser_jobs.status('XXX').get(a='b')
+            expect(self.client.get).to(have_been_called_with(
+                '/v2/travel/trip-parser-jobs/XXX', a='b'
+            ))
+
+        with it('.travel.trip_parser_jobs.result().get'):
+            self.client.travel.trip_parser_jobs.result('XXX').get(a='b')
+            expect(self.client.get).to(have_been_called_with(
+                '/v2/travel/trip-parser-jobs/XXX/result', a='b'
```
Comment (anthonyroux) [inline specs/namespaces/namespaces_spec.py]: Shouldn't we have tests for the post as well?

**Pull request merged**